### PR TITLE
Search with a / in the title

### PR DESCRIPTION
**Pull request by brianteeman**
PR for #18220

Code by @hikashop-nicholas

### Steps to reproduce the issue

Create a new article with the title "SHIM/SET"
Search for it on your frontend via the Joomla search.

### Expected result

The result page should display a search result for it with a link to the article.

### Actual result

The result page has a search result but without any link.

> Warning: preg_replace(): Unknown modifier 't' in C:\Apache24\htdocs\cms\components\com_search\views\search\view.html.php on line 268

### Test
Apply this PR and the title is displayed

### CODE REVIEW

**Comment by eXsiLe95** [comment]
I have tested this item :red_circle: unsuccessfully on b883cf5dfd37a5a7dac9f31e1c0ffe50d2105814<br /><br />
### System Information

* Running on xampp v3.2.2
* Windows 10 (1703: 15063:540)
* Chrome 61.0.3163.100 (64bit)
* PHP 7.1.7
* Joomla! 3.8.1-rc2-dev

### Testing instructions

1. Fresh installation of Joomla! staging
1. Create any article with Slash in name
1. Search for created article

### Before patch

An error occured, article was shown without headline.


### With patch

Still, I can't search for any article with a slash "/" in the title. In my example, I created an article THM/MNI.

localhost-bugtesting-index php-search-searchword thm-mni searchphrase all



Tested @iCampus<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/18221">issues.joomla.org/tracker/joomla-cms/18221</a>.</sub>

File: components/com_search/views/search/view.html.php [hunk 1]
```diff
@@ -265,6 +265,9 @@ public function display($tpl = null)
 					$created = JHtml::_('date', $result->created, JText::_('DATE_FORMAT_LC3'));
 				}
 
+				// Make sure there are no slashes in the needle
+			//	$needle = str_replace('/','\/',$needle);
```
Comment (eXsiLe95) [inline components/com_search/views/search/view.html.php]: Shouldn't this be uncommented?

Comment (brianteeman) [reply components/com_search/views/search/view.html.php]: oops

**Comment by brianteeman** [comment]
@eXsiLe95 please can you retest - silly mistake by me leaving the comment in

**Comment by eXsiLe95** [comment]
I have tested this item :white_check_mark: successfully on b883cf5dfd37a5a7dac9f31e1c0ffe50d2105814<br /><br />

No problem!

After the quick fix, it now works as expected! (for details to testing, see post above)

image


<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/18221">issues.joomla.org/tracker/joomla-cms/18221</a>.</sub>

File: components/com_search/views/search/view.html.php [hunk 2]
```diff
@@ -265,6 +265,9 @@ public function display($tpl = null)
 					$created = JHtml::_('date', $result->created, JText::_('DATE_FORMAT_LC3'));
 				}
 
+				// Make sure there are no slashes in the needle
+				$needle = str_replace('/','\/',$needle);
+
```
Comment (Quy) [inline components/com_search/views/search/view.html.php]: Add space after commas. Remove tabs in line 270.

Comment (brianteeman) [reply components/com_search/views/search/view.html.php]: done

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on 3aa0983f97f495ddb14a5422614cd5bd389d6f98<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/18221">issues.joomla.org/tracker/joomla-cms/18221</a>.</sub>

**Comment by ghost** [comment]
@eXsiLe95 can you please retest?

**Comment by Quy** [comment]
@franz-wohlkoenig It is only coding style so it would not affect the test.

**Comment by ghost** [comment]
RTC after two successful tests.

**Comment by eXsiLe95** [comment]
I have tested this item :white_check_mark: successfully on 3aa0983f97f495ddb14a5422614cd5bd389d6f98<br /><br />I also tested this one. Works as expected. Didn't expect it not to work after code style changes, too.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/18221">issues.joomla.org/tracker/joomla-cms/18221</a>.</sub>

**Comment by brianteeman** [comment]
thanks

**Pull request merged**